documentation/website/src/handbook/integration/serenity-2-migration-guide.md

```diff
@@ -0,0 +1,697 @@
+[flaky](https://testing.googleblog.com/2016/05/flaky-tests-at-google-and-how-we.html).
+  
+The code for `Hide` is now shown using both Serenity/JS 1.x and Serenity/JS 2.x implementations:
+
+### Serenity/JS 1.x implementation of Hide Task
+
+```javascript
+import { PerformsTasks, Task, UsesAbilities } from "serenity-js/lib/screenplay";
+import { BrowseTheWeb, Target } from "serenity-js/lib/serenity-protractor";
+
+export class Hide implements Task {
+
+    static target(target: Target): Task {
+        return new Hide(target);
+    }
+
+    constructor(private target: Target) {                                                            (1)
+    }
+
+    performAs(actor: PerformsTasks & UsesAbilities): PromiseLike<void> {
+        const browseTheWeb = BrowseTheWeb.as(actor);                                                 (2)
+        const element = browseTheWeb.locate(this.target);                                            (3)
+        return element.isPresent()
+         .then(present =>                                                                            (4)
+          present && browseTheWeb.executeScript("arguments[0].style.visibility='hidden'", element)); (5)
+    }
+}
+```
+In the example above:
+ 1. the `Target` was passed in the constructor.
+ 2. the `BrowseTheWeb` ability was extracted to a local variable as it will be used more than once in the function body.
+ 3. a `WebElement`is resolved by passing the `Target` to the `locate` method.
+ 4. the `isPresent()` method is called on the `WebElement` and a `present` value is extracted in the `Promise` `then` block.
+ 5. if the `present` value is `true` the `executeScript` method is executed. Two arguments are passed: 
+    - the script text 
+    - the web element
+
+### Serenity/JS 2.x implementation of Hide Task
+
+```javascript
+import { AnswersQuestions, PerformsActivities, Question, Task, UsesAbilities } from "@serenity-js/core";
+import { ExecuteScript } from "@serenity-js/protractor";
+import { promiseOf } from "@serenity-js/protractor/lib/promiseOf";
+import { ElementFinder } from "protractor";
+
+export class Hide implements Task {
+
+  static target(target: Question<ElementFinder> | ElementFinder): Task {
+    return new Hide(target);
+  }
+
+  constructor(private target: Question<ElementFinder> | ElementFinder) {                        (1)
+  }
+
+  performAs(actor: AnswersQuestions & PerformsActivities & UsesAbilities): Promise<void> {
+    return promiseOf(this.target.answeredBy(actor).isPresent()                                  (2)(6)
+      .then(present =>                                                                          (3)
+       present && actor.attemptsTo(ExecuteScript.sync(`arguments[0].style.visibility='hidden'`) (4)
+       .withArguments(this.target))));                                                          (5)
+  }
+}
```

Comment:
I wonder if it would be cleaner to implement it as follows:

```
export const Hide = (target: Question<ElementFinder> | ElementFinder) =>
    Task.where(`#actor hides ${ target }`,
        Check.whether(target, isPresent()).
            andIfSo(
                ExecuteScript.sync(`arguments[0].style.visibility='hidden'`).withArguments(target),
            ),
    );
```

Reply:
👍 done